### ISSUE TITLE
munin-update fails in 2.0.68 with 'Can't use string ("local_address") as a HASH ref while "strict refs" in use'

### ISSUE THREAD

**Issue by malteger** [munin-monitoring/munin#1440]
**Describe the bug**

After updating from 2.0.67 to 2.0.68 munin-update fails with
```
[ERROR] Munin::Master::UpdateWorker<...> died with 'Can't use string ("local_address") as a HASH ref while "strict refs" in use at /usr/share/perl5/vendor_perl/Munin/Master/Node.pm line 158.'
 ```

After reverting de74370848a7d18ce780dfcf8dbd8dbb026f6128 and 64794bed4cedf8d95d0d3f8845d08399fdddd4f0 everything is working again.

Another way to fix it was to change `_get_node_or_global_setting` to `$self->_get_node_or_global_setting` as in the TLS section. As I have knowledge of Perl this might not be the correct solution, but if it is I can provide a PR.

**To Reproduce**
1. Update from 2.0.67 to 2.0.68
2. Nothing is updated anymore

**Screenshots & Logs**
```
2021/11/21 00:05:01 [INFO]: Starting munin-update
2021/11/21 00:05:02 [INFO] Process 16030 is dead, stealing lock, removing file
2021/11/21 00:05:02 [INFO] Process 16029 is dead, stealing lock, removing file
2021/11/21 00:05:02 [INFO] starting work in 27039 for XXX (127.0.0.1:4949).
2021/11/21 00:05:02 [INFO] Process 16031 is dead, stealing lock, removing file
2021/11/21 00:05:02 [ERROR] Munin::Master::UpdateWorker<XXX> died with 'Can't use string ("local_address") as a HASH ref while "strict refs" in use at /usr/share/perl5/vendor_perl/Munin/Master/Node.pm line 158.'
2021/11/21 00:05:02 [INFO] starting work in 27041 for XXX (10.8.0.1:4949).
2021/11/21 00:05:02 [INFO] starting work in 27040 for XXX (127.0.0.1:4949).
2021/11/21 00:05:02 [ERROR] Munin::Master::UpdateWorker<XXX> died with 'Can't use string ("local_address") as a HASH ref while "strict refs" in use at /usr/share/perl5/vendor_perl/Munin/Master/Node.pm line 158.'
2021/11/21 00:05:02 [ERROR] Munin::Master::UpdateWorker<XXX> died with 'Can't use string ("local_address") as a HASH ref while "strict refs" in use at /usr/share/perl5/vendor_perl/Munin/Master/Node.pm line 158.'
2021/11/21 00:05:12 [INFO] Remaining workers: XXX
2021/11/21 00:05:12 [INFO] Reaping Munin::Master::UpdateWorker<XXX>.  Exit value/signal: 18/0
2021/11/21 00:05:12 [INFO] Reaping Munin::Master::UpdateWorker<XXX>.  Exit value/signal: 18/0
2021/11/21 00:05:12 [INFO] Reaping Munin::Master::UpdateWorker<XXX>.  Exit value/signal: 18/0
2021/11/21 00:05:12 [INFO]: Munin-update finished (10.43 sec)
```

**Desktop (please complete the following information):**
 - Arch Linux ARM
 - Munin Version 2.0.68

**Comment by yason** [comment]
Hi!
The same error on OpenBSD after upgrading to 2.0.68

**Comment by sumpfralle** [comment]
Thank you for your report!

> As I have knowledge of Perl this might not be the correct solution [..]

Just for my understanding (since my perl-foo is weak): do you think, this would be the proper solution or are you unsure about this?

**Comment by sumpfralle** [comment]
> Additionally, (and maybe unrelated) the commits replace defined with exists in the lookup behaviour, which might change the resulting value if the hash contains undef according to a quick google search.

Yes, this is slightly different, but should be fine for these hashes here. Thank you for taking a closer look!

I created a pull request for fixing the issue: #1442.

Could someone of you please test, whether it fixes the issue for you?

**Comment by sumpfralle** [comment]
The issue is fixed now via 291bfe912c600b0c98a72c7d2684de8f8249bc76.
@malteger: thanks for researching and reporting the issue and for testing the fix!
